### PR TITLE
Feature/n21 98 invalidate nextcloud session

### PR DESCRIPTION
**Pull request by alweber-cap**
# Description

Added Global Logout URI and New Hydra Env-Vars for Logout Flow

## Links to Tickets or other pull requests

https://ticketsystem.dbildungscloud.de/browse/N21-93
hpi-schul-cloud/schulcloud-client#2865
hpi-schul-cloud/schulcloud-server#3469

## Approval for review

- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done

More and detailed information on the _definition of done_ can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

**Review by mamutmk5 — CHANGES_REQUESTED** [review]

File: ansible/group_vars/nbc/instance_cfg.yml [hunk 1]
```diff
@@ -83,6 +83,8 @@ ES_MERLIN_AUTH_URL: http://merlin.nibis.de/auth.php
 PROD_INSTANCE: true
 # Matrix
 RABBITMQ_MATRIX_QUEUE_EXTERNAL: matrix_sync_populated2
+#Hydralogout
+HYDRA_LOGOUT_URI: http://hydra-svc:4444/oauth2/sessions/logout
```
[the remark this answers is not in the document]
Comment (alweber-cap) [reply ansible/group_vars/nbc/instance_cfg.yml:87]: will change it to external route found in hydra ingress

**Review by MarvinOehlerkingCap — APPROVED** [review]

File: ansible/group_vars/nbc/instance_cfg.yml [hunk 2]
```diff
@@ -83,6 +83,8 @@ ES_MERLIN_AUTH_URL: http://merlin.nibis.de/auth.php
 PROD_INSTANCE: true
 # Matrix
 RABBITMQ_MATRIX_QUEUE_EXTERNAL: matrix_sync_populated2
+#Hydralogout
+HYDRA_LOGOUT_URI: "https://{{ HYDRA_DNS_PREFIX }}{{ DOMAIN }}/oauth2/sessions/logout"
```
Comment (mamutmk5) [inline ansible/group_vars/nbc/instance_cfg.yml:87]: Why don't you set it as default for all instances even they don't have nextcloud jet?

Comment (alweber-cap) [reply ansible/group_vars/nbc/instance_cfg.yml:87]: The existence of this ENVVAR is also used as a featuretoggle for the oauth logout. If there is no problem with setting it as a global default (we tested it and it doesn't affect functionality) we can do that

Comment (mamutmk5) [reply ansible/group_vars/nbc/instance_cfg.yml:87]: Both are okay

File: ansible/group_vars/all/config.yml [hunk 3]
```diff
@@ -733,3 +733,7 @@ configuration_all:
     server: false
     client: true
     nuxtclient: false
+  HYDRA_LOGOUT_URI:
+    server: false
+    client: true
+    nuxtclient: false
```
Comment (mamutmk5) [inline ansible/group_vars/all/config.yml:739]: I struggle a bit with the name _HYDRA_LOGOUT_URI_ for me it's feel like it should be more general like _OAUTH_LOGOUT_URI_ or so

Comment (Loki-Afro) [reply ansible/group_vars/all/config.yml:739]: i agree :) - is there anything speaking against it?

**Review by mamutmk5 — APPROVED** [review]

**Pull request merged**